Feature/native file and piece model class

erikap opened the pull request:
Convert case-type, piece and file model to native classes.

erikap committed:
Refactor case-type model to native ES6 class

erikap committed:
Refactor file model to native ES6 class

erikap committed:
Refactor piece model to native ES6 class

erikap committed:
Add canPreview property on file model

Flag to indicate whether the file can be opened in the browser for
preview. Currently only PDF is supported.

MikiDi commented on app/models/file.js:
```diff
@@ -1,46 +1,45 @@
+    return this.filename.replace(regex, '');
+  }
+
+  get canPreview() {
```
Ik zie dat je deze test naar het model refactored hebt, maar hiervan zou ik durven argumenteren dat het toch eerder een eigenschap van de view-component is dan van het model (hoewel dit in de praktijk nog niet voorvalt, zou je je kunnen inbeelden dat 1 component enkel pdf's ondersteunt, maar een andere bv. ook word-files)

erikap replied:
Ik zat er vooral mee verveeld dat die code nu op 2 plaatsen gedupliceerd was en de relation die erin opgehaald wordt niet ge-await werd.

Je hebt gelijk dat er in de toekomst meer viewers kunnen zijn. Ik begin daarom meer te denken richting verantwoordelijkheid van de view-component zelf ipv de route/controller errond. Aangezien we die `PdfView`-component waarschijnlijk niet afhankelijk willen maken van ons file data model, misschien een wrapping component `DocumentView` waaraan we `@file` meegegeven en die dan zelf beslist met welke viewer hij het document toont (in dit geval dus PDF of niets, maar kan uitgebreid worden met andere viewers in de toekomst). Wat denk je?

MikiDi replied:
Akkoord voor `DocumentView`. Zo was het oorspronkelijk trouwens ook voorzien :-) , maar omdat dat voor de eerste tijd een onnodig niveau van indirectie was, is dat er terug uit gegaan ...

erikap committed:
Merge branch 'feature/KAS-2835_authorised_document_view' into feature/native-file-and-piece-modal-class

erikap committed:
Add document view component to render document preview

The document view component contains the logic to determine which
renderer must be used to display the document. Currently only PDF
files are supported.
